core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -483,15 +483,20 @@ private void failEarly(Throwable cause) {
 
     // TODO(ikhoon): Consider moving the logic for filling authority to `HttpClientDelegate.exceute()`.
     private void autoFillSchemeAuthorityAndOrigin() {
-        final String authority = authority();
-        if (authority != null && endpoint != null && endpoint.isIpAddrOnly()) {
-            // The connection will be established with the IP address but `host` set to the `Endpoint`
-            // could be used for SNI. It would make users send HTTPS requests with CSLB or configure a reverse
-            // proxy based on an authority.
-            final String host = SchemeAndAuthority.of(null, authority).host();
-            if (!NetUtil.isValidIpV4Address(host) && !NetUtil.isValidIpV6Address(host)) {
-                endpoint = endpoint.withHost(host);
+
+        try {
+            final String authority = authority();
+            if (endpoint != null && endpoint.isIpAddrOnly()) {
+                // The connection will be established with the IP address but `host` set to the `Endpoint`
+                // could be used for SNI. It would make users send HTTPS requests
+                // with CSLB or configure a reverse proxy based on an authority.
+                final String host = SchemeAndAuthority.of(null, authority).host();
+                if (!NetUtil.isValidIpV4Address(host) && !NetUtil.isValidIpV6Address(host)) {
+                    endpoint = endpoint.withHost(host);
+                }
             }
+        } catch (IllegalStateException e) {
+            // Just pass, because it's normal condition.
```

Comment:
What do you think of creating a private method to a nullable value instead of catching the exception?
```java
@Nullable
private String authorityOrNull();
```

Reply:
Wow, It is better way! 
@ikhoon nim, I have a question. 
Should the methods `authority()` and `host()` throw a `CheckedException` so that the user is forced to handle the error?

IMHO, I think `UncheckedException` is better than `CheckedException` in case. because it's easy to read.